projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java

```diff
@@ -399,125 +439,155 @@ else if (bindaasConfigurationProtocol.equals("JWT") && protocol.equals("jwt")) {
 					case TOKEN_ALREADY_EXIST:
 						return exceptionToResponse(new DuplicateJWTException(
 								username), applicationID, "not-resolved", username);
-					case TOKEN_DOES_NOT_EXIST:
-						return exceptionToResponse(new JWTDoesNotExistException(
-								username), applicationID, "not-resolved", username);
 					default:
-						JsonObject retVal = new JsonObject();
-						retVal.add("error",new JsonPrimitive(JWTManagerException.getMessage()));
-						return Response.status(500).entity(retVal.toString()).
-								type("application/json").build();
+						return exceptionToResponse(JWTManagerException.getMessage());
 				}
 			} catch (Exception e) {
-				JsonObject retVal = new JsonObject();
-				retVal.add("error",new JsonPrimitive(e.getMessage()));
-				return Response.status(500).entity(retVal.toString()).
-						type("application/json").build();
+				return exceptionToResponse(e.getMessage());
 			}
 		}
 		else {
-			JsonObject retVal = new JsonObject();
-			retVal.add("error",new JsonPrimitive("Authentication protocol specified does not match with server's configuration"));
-			return Response.status(500).entity(retVal.toString()).
-					type("application/json").build();
+			return exceptionToResponse(protocolDoesNotMatchServerConfiguration);
 		}
 
 	}
 
 	@Override
 	@DELETE
 	@Path("/revokeUser")
-	public Response revokeAccess(@HeaderParam("_username") String username,
+	public Response revokeAccess(@HeaderParam("_protocol") String protocol,
+			@HeaderParam("_username") String username,
 			@HeaderParam("_applicationID") String applicationID,
 			@HeaderParam("_salt") String salt,
 			@HeaderParam("_digest") String digest,
 			@QueryParam("comments") String comments) {
-		try {
-			TrustedApplicationEntry trustedAppEntry = authenticateTrustedApplication(
-					applicationID, salt, digest, username);
+		// FIXME enforce strict if else condition here?
+		if(bindaasConfigurationProtocol.equals("API_KEY") && protocol.equals("api_key")){
+			try {
+				TrustedApplicationEntry trustedAppEntry = authenticateTrustedApplication(
+						applicationID, salt, digest, username);
 
-			comments = comments == null ? comments = "API Key Revoked via Trusted Application API"
-					: comments;
-			Integer count = apiKeyManager.revokeAPIKey(
-					new BindaasUser(username), trustedAppEntry.getName(),
-					comments, ActivityType.REVOKE);
+				comments = comments == null ? comments = "API Key Revoked via Trusted Application API"
```

Comment:
Shall we simplify this as:

if (comments == null) {
comments = "API Key Revoked via Trusted Application API";
}

(Hope that is what you are trying to do here).